apps/src/templates/progress/GoogleClassroomShareButton.jsx

```diff
@@ -0,0 +1,108 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Button from '../Button';
+import i18n from '@cdo/locale';
+
+// https://developers.google.com/classroom/brand
+const styles = {
+  label: {
+    paddingLeft: 16,
+    textAlign: 'left'
+  },
+  container: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'start'
+  }
+};
+
+const GooglePlatformApiId = 'GooglePlatformApiId';
+
+class GoogleClassroomShareButton extends React.Component {
+  static propTypes = {
+    buttonId: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+    itemtype: PropTypes.string.isRequired,
+    title: PropTypes.string,
+    height: PropTypes.number,
+    courseid: PropTypes.number
+  };
+
+  constructor(props) {
+    super(props);
+    this.loadApi = this.loadApi.bind(this);
+    this.waitForGapi = this.waitForGapi.bind(this);
+    this.renderButton = this.renderButton.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.gapiReady()) {
+      this.renderButton();
+    } else {
+      this.loadApi();
+    }
+  }
+
+  /* The Google Classroom Share Button is only available through their js api,
+   * so we have to add it to the page on load. Each instance of this
+   * component will wait until the api is ready before rendering the button,
+   * but only the first will kick of the loading process.
+   */
+  loadApi() {
+    if (!document.getElementById(GooglePlatformApiId)) {
+      window.___gcfg = {
+        parsetags: 'explicit'
+      };
+
+      const gapi = document.createElement('script');
+      gapi.src = 'https://apis.google.com/js/platform.js';
+      gapi.id = GooglePlatformApiId;
+      gapi.addEventListener('load', this.waitForGapi);
+      document.body.appendChild(gapi);
+    } else {
+      this.waitForGapi();
+    }
+  }
+
+  waitForGapi() {
+    if (this.gapiReady()) {
+      this.renderButton();
+    } else {
+      setTimeout(() => {
+        this.waitForGapi();
+      }, 100);
+    }
+  }
+
+  gapiReady() {
+    return !!window.gapi && typeof window.gapi.sharetoclassroom !== 'undefined';
+  }
+
+  // https://developers.google.com/classroom/guides/sharebutton
+  renderButton() {
+    window.gapi.sharetoclassroom.render(this.props.buttonId, {
+      theme: 'light',
+      url: this.props.url,
+      itemtype: this.props.itemtype,
+      title: this.props.title,
+      size: this.props.height,
+      courseid: this.props.courseid
+    });
+  }
+
+  render() {
+    return (
+      <span style={styles.container}>
+        <span id={this.props.buttonId} />
+        <span style={styles.label}>{i18n.shareToGoogleClassroom()}</span>
+      </span>
+    );
+  }
+}
+
+GoogleClassroomShareButton.defaultProps = {
+  itemtype: 'assignment',
+  height: Button.ButtonHeight.default
+};
+
+export default GoogleClassroomShareButton;
```

Comment:
NIT: This isn't in our style guide, but I always see us do `export default` at the class declaration. The exception to this is when we are using redux or radium. 
My two cents: In general, I've found this to be a useful convention so I know whether the component is using redux/radium and so I have to explicitly make the decision to include multiple components in a file.

Reply:
same as above... we don't seem to be very consistent about this and a lot of the components i've worked with are doing `export default connect` at the bottom of the file, but for non-redux components the convention you suggest makes sense.

Reply:
I should say that for this one, I think this might just be my personal preference (so not asking you to change it if you don't want to). If you have thoughts either way, I'd love to hear them!

Reply:
for things like this i'm not very opinionated, other than the opinion that i like consistency. i put your preference in the style guide proposal and we can see if anyone has a stronger preference in the other direction!